crates/ruma-events/src/room/power_levels.rs

```diff
@@ -437,11 +437,18 @@ impl RoomPowerLevels {
         acting_pl >= self.kick && target_pl < acting_pl
     }
 
-    /// Whether the given user can redact events based on the power levels.
+    /// Whether the given user can redact their own events based on the power levels.
     ///
-    /// Shorthand for `power_levels.user_can_do(user_id, PowerLevelAction::Redact)`.
-    pub fn user_can_redact(&self, user_id: &UserId) -> bool {
-        self.for_user(user_id) >= self.redact
+    /// Shorthand for `power_levels.user_can_do(user_id, PowerLevelAction::RedactOwn)`.
+    pub fn user_can_redact_own_event(&self, user_id: &UserId) -> bool {
+        self.user_can_send_message(user_id, MessageLikeEventType::RoomRedaction)
```

Comment:
Sooo.. One can set power levels such that redacting one's own messages is not allowed (for ~everyone)? Does that work in practice (with Synapse)? 😅

Reply:
It should work with Synapse, it's [explicitely said in the spec](https://spec.matrix.org/v1.9/client-server-api/#put_matrixclientv3roomsroomidredacteventidtxnid).